### PR TITLE
Feature/omer/menu keyboard navigation

### PR DESCRIPTION
**Pull request by omerdoron1**
Add keyboard navigation for menu and sub menus

### CODE REVIEW

File: src/hooks/useKeyEvent.js [hunk 1]
```diff
@@ -15,38 +16,39 @@ export default function useKeyEvent({
   const documentRef = useRef(document);
   const onKeyUpPress = useCallback(
     event => {
+      const { key } = event;
+      if (!keys.includes(key)) {
+        return;
+      }
+
+      name && console.log("evnt for  ", name, " preventDefault: ", preventDefault);
```
Comment (orrgottlieb) [inline src/hooks/useKeyEvent.js:24]: why name?

File: src/hooks/useIsMouseEnter.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useCallback, useState } from "react";
```
Comment (orrgottlieb) [inline src/hooks/useIsMouseEnter.js:1]: Cool!

File: src/components/Menu/Menu/Menu.jsx [hunk 3]
```diff
@@ -1,69 +1,144 @@
-import React, { useState, useRef, useCallback } from "react";
+import React, { forwardRef, useLayoutEffect, useState, useRef, useCallback } from "react";
 import PropTypes from "prop-types";
 import cx from "classnames";
+import useMergeRefs from "../../../hooks/useMergeRefs";
 import useKeyEvent from "../../../hooks/useKeyEvent";
 import { MENU_SIZES } from "./MenuConstants";
 import "./Menu.scss";
 
+export const usePrevious = value => {
```
Comment (orrgottlieb) [inline src/components/Menu/Menu/Menu.jsx:9]: could move to hooks?

File: src/components/Menu/Menu/Menu.jsx [hunk 4]
```diff
@@ -10,60 +11,132 @@ const isChildSelectable = (newIndex, children) => {
...
 
-const Menu = ({ classname, size, tabIndex, ariaLabel, children }) => {
-  const ref = useRef(null);
-  const [activeItemIndex, setActiveItemIndex] = useState(-1);
+const Menu = forwardRef(
+  (
+    { id, classname, size, tabIndex, ariaLabel, children: originalChildren, isVisible = true, closeSubMenu },
+    forwardedRef
+  ) => {
+    const children = React.Children.toArray(originalChildren);
```
Comment (orrgottlieb) [inline src/components/Menu/Menu/Menu.jsx:19]: Can we use useMemo here?

**Review by orrgottlieb — APPROVED** [review]

File: src/components/Menu/Menu/Menu.jsx (hunk 4 above)
Comment (omerdoron1) [reply src/components/Menu/Menu/Menu.jsx]: done :)

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**